lib/AST/ModuleNameLookup.cpp

```diff
@@ -278,6 +280,43 @@ void ModuleNameLookup<LookupStrategy>::lookupInModule(
   if (decls.size() - initialCount <= 1)
     return;
 
+  // Some functions like `memchr` are defined both in libc and in libc++.
+  // Importing both would result in ambiguities, but there are some attributes
+  // that mark the preferred overloads. See _LIBCPP_PREFERRED_OVERLOAD.
+  llvm::SmallPtrSet<ValueDecl *, 4> declsToRemove;
+  bool hasPreferredOverload = false;
+  for (auto decl : decls)
+    if (const auto *clangDecl = decl->getClangDecl()) {
+      if (clangDecl->hasAttr<clang::EnableIfAttr>()) {
+        // FIXME: at some point we might want to call into Clang to implement
+        // the full enable_if semantics including the constant evaluation of the
+        // conditions. For now, just look for the first enable_if(true, "...")
+        // and assume all the rest of the enable_ifs evaluate to true.
+        bool thisDeclHasPreferredOverload = false;
+        for (auto clangAttr :
+             clangDecl->specific_attrs<clang::EnableIfAttr>()) {
+          if (auto litExpr =
+                  dyn_cast<clang::CXXBoolLiteralExpr>(clangAttr->getCond())) {
+            if (litExpr->getValue()) {
+              thisDeclHasPreferredOverload = hasPreferredOverload = true;
+              break;
+            }
+          }
+        }
+        if (!thisDeclHasPreferredOverload)
+          declsToRemove.insert(decl);
+      } else
+        declsToRemove.insert(decl);
```

Comment:
@Xazax-hun Does this implementation match the PR comment of:

```
This PR modifies the lookup logic to exclude the non-preferred Clang functions from the overload set whenever a preferred overload is available.
```

?

Reading the implementation it looks like when there are no decls with `_LIBCPP_PREFERRED_OVERLOAD`, **every** `clangDecl` in the `decls` will be added to `declsToRemove` and thus later removed from `decls`. The PR description makes it sound like there's only a behavior change when `_LIBCPP_PREFERRED_OVERLOAD` is present on a `clangDecl` but the implementation makes it looks like all the decls will be removed. Or am I misreading this?